avocado/core/spawners/common.py

```diff
@@ -51,8 +52,9 @@ def stream_output(self, job_id, task_id):
         one is not suitable for your spawner. i.e: if the spawner is trying to
         access a remote output file.
         """
-        src = '~/avocado/job-results/{}/test-results/{}/data'.format(job_id,
-                                                                     task_id)
+        results_dir = get_job_results_dir(job_id)
+        src = '{}/test-results/{}/data'.format(results_dir,
+                                               task_id.replace('/', '_'))
```

Comment:
Maybe use: https://avocado-framework.readthedocs.io/en/79.0/api/utils/avocado.utils.html#avocado.utils.astring.string_to_safe_path

This is already used in the avocado.core.test.TestID.str_filesystem.

Reply:
Yes, sure. Will send this suggestion with v2.